File: DataBinding/app/src/main/java/com/example/databinding/Adapter.java
package com.example.databinding;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.example.databinding.databinding.ItemLayoutBinding;

import java.util.ArrayList;
import java.util.List;

public class Adapter extends RecyclerView.Adapter<Adapter.MyViewHolder> {

    private Context context;
    private List<User> users= new ArrayList<>();

    public Adapter(Context context, List<User> users) {
        this.context = context;
        this.users = users;
    }


    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemLayoutBinding itemLayoutBinding = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()),
                R.layout.item_layout,parent,false);

        MyViewHolder myViewHolder = new MyViewHolder(itemLayoutBinding);

        return myViewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        User user = users.get(position);
        holder.itemLayoutBinding.setUser(user);
    }

    @Override
    public int getItemCount() {
        return users.size();
    }

    public class MyViewHolder extends RecyclerView.ViewHolder {
        ItemLayoutBinding itemLayoutBinding;
        public MyViewHolder(@NonNull ItemLayoutBinding itemView) {
            super(itemView.getRoot());
            itemLayoutBinding = itemView;
        }
    }
}

File: ImageProject/app/src/main/java/com/example/imageproject/Adapter.kt
package com.example.imageproject

import android.content.Context
import android.graphics.Bitmap
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.item_layout.view.*

class Adapter(val iList : List<Bitmap>, val context : Context) : RecyclerView.Adapter<Myviewholder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Myviewholder {
        val view = LayoutInflater.from(context).inflate(R.layout.item_layout,parent,false)
        return Myviewholder(view)
    }

    override fun getItemCount(): Int {
        return iList.size
    }

    override fun onBindViewHolder(holder: Myviewholder, position: Int) {
        holder.rImage.setImageBitmap(iList[position])
    }
}

class Myviewholder(itemview: View) : RecyclerView.ViewHolder(itemview){
    val rImage = itemView.img_item
}
